static/index.js

```diff
@@ -0,0 +1,294 @@
+        key = "jump";
+        break;
+    }
+    if (key != "") {
+      keylogger(keys);
+      send({
+        action: "move",
+        botId: getCurrentBot(),
+        data: { operation: key, state: false },
+      });
+    }
+  },
+  false
+);
+
+// looks like a function but is actually just to trigger the listeners after the server is online
+function listenSocket() {
+  socketserver.onopen = function (event) {
+    console.log("connected");
+    connected = true;
+    serverOnline(true);
+  };
+  socketserver.onmessage = function (event) {
+    console.log(event.data);
+    var message = JSON.parse(event.data);
+    switch (message.action) {
+      case "coords":
+        setCoords(message);
+        break;
+      case "health":
+        setHealth(message);
+        break;
+      case "hunger":
+        setHunger(message);
+        break;
+      case "started":
+        botOnline(message.action);
+        break;
+    }
+  };
+  socketserver.onclose = function (event) {
+    console.log("connection closed");
+    serverOnline(false);
+  };
+}
+
+// functions
+
+function say(message) {
+  send({
+    action: "say",
+    botId: getCurrentBot(),
+    message: message,
+  });
+}
+function kill() {
+  send({
+    action: "kill",
+    botId: getCurrentBot(),
+  });
+}
+
+function place() {
+  send({
+    action: "place",
+    botId: getCurrentBot(),
+  });
+}
+function attack() {
+  send({
+    action: "attack",
+    botId: getCurrentBot(),
+  });
+}
+function setServer(address, port) {
+  var address = document.getElementById("server").value;
+  var port = document.getElementById("port").value;
+  var data = { address: address, port: port };
+
+  send({
+    action: "setServer",
+    botId: getCurrentBot(),
+    data: data,
+  });
+}
+function stopBot() {
+  send({
+    action: "stop",
+    botId: getCurrentBot(),
+  });
+}
+function startBot() {
+  send({
+    action: "start",
+    botId: getCurrentBot(),
+  });
+}
+function getCoords() {
+  if (connected) {
+    send({
+      action: "getCoords",
+      botId: getCurrentBot(),
+    });
+  }
+}
+function getHealth() {
+  if (connected) {
+    send({
+      action: "getHealth",
+      botId: getCurrentBot(),
+    });
+  }
+}
+function getHunger() {
+  if (connected) {
+    send({
+      action: "getHunger",
+      botId: getCurrentBot(),
+    });
+  }
+}
+
+function setHunger(message) {
+  document.getElementById("hunger").innerHTML = message.data;
+}
+
+function setCoords(message) {
+  document.getElementById("coords").innerHTML =
+    Math.round(message.data.x) +
+    " " +
+    Math.round(message.data.y) +
+    " " +
+    Math.round(message.data.z);
+}
+function setHealth(message) {
+  document.getElementById("health").innerHTML = message.data;
+}
+
+function getCurrentBot() {
+  return document.getElementById("botSelected").value;
+}
+
+/*
+required headers
+action:
+botId:
+
+*/
+function send(message) {
+  JSON.stringify(message);
+  socketserver.send(JSON.stringify(message));
+}
```

Comment:
This isn't being used anywhere - probably could be removed.

Reply:
i’m pretty sure it is

Reply:
Oh yeah. My bad